### PR TITLE
Implement `Widget.presence_count` attribute and fix `Widget` types

### PR DESCRIPTION
**Pull request by m-etroo**
## Summary

 * Implements missing `Widget.presence_count` attribute
 * Fixes the return types of `Widget.invite_url` and `Widget.fetch_invite` to make them Optional
 * Removes `_WidgetOptional` since none of the widget object fields returned by the API are optional
 * Minor documentation changes

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/types/widget.py [hunk 1]
```diff
@@ -49,13 +49,13 @@ class WidgetMember(User, total=False):
...
     channels: List[WidgetChannel]
     members: List[WidgetMember]
-    presence_count: int
 
 
 class Widget(_WidgetOptional):
     id: Snowflake
     name: str
     instant_invite: str
+    presence_count: int
```
Comment (Rapptz) [inline discord/types/widget.py:58]: The `instant_invite` type needs to be `Optional[str]`.

Comment (m-etroo) [reply discord/types/widget.py:58]: Thanks. Is there a reason why `channels` and `members` are left in the `_WidgetOptional` class? They are not listed as optional fields on the API docs.

Comment (Rapptz) [reply discord/types/widget.py:58]: Those can probably be moved too, assuming the documentation doesn't lie (dangerous assumption but oh well).

**Review by Rapptz — APPROVED** [review]
Thanks.

**Review by Rapptz — CHANGES_REQUESTED** [review]
Sorry, I just realised this is a breaking change.

This needs to be documented in `migrating.rst`.

**Review by Rapptz — APPROVED** [review]
Thanks again.

**Pull request merged**